tests/testsuite/build.rs

```diff
@@ -244,6 +244,33 @@ fn cargo_compile_directory_not_cwd() {
     assert!(p.bin("foo").is_file());
 }
 
+#[cargo_test]
+fn cargo_compile_with_unsupported_short_unstable_feature_flag() {
+    let p = project()
+        .file("Cargo.toml", &basic_bin_manifest("foo"))
+        .file("src/foo.rs", &main_file(r#""i am foo""#, &[]))
+        .file(".cargo/config.toml", &"")
+        .build();
+
+    p.cargo("-zunstable-options -C foo build")
+        .masquerade_as_nightly_cargo(&["chdir"])
+        .cwd(p.root().parent().unwrap())
+        .with_stderr(
+            "\
+error: unexpected argument '-z' found
+
+  tip: a similar argument exists: '-Z'
+
+Usage: cargo [+toolchain] [OPTIONS] [COMMAND]
```

Comment:
Is displaying `cargo[EXE]` on Windows necessary?

Reply:
Its necessary with clap when using the default generated usage because it uses `argv[0].file_name()`.  However, `cargo` overrides the usage to include the `[+toolchain]` and `<MANIFEST_RS>`

Reply:
I meant do we need to display it to get the same experience with clap?

Reply:
Likely not worth it.